### PR TITLE
fix most compile warnings

### PR DESCRIPTION
**Pull request by MatthewBemis**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala [hunk 1]
```diff
@@ -101,13 +101,6 @@ trait UserRoutes extends UserInfoDirectives {
                 }
               } ~
               pathPrefix("petServiceAccount") {
-                pathEndOrSingleSlash {
```
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala:104]: nothing should be using this. it has been deprecated and hidden from swagger for months

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/schema/JndiSchemaDAO.scala [hunk 2]
```diff
@@ -73,10 +73,12 @@ class JndiSchemaDAO(protected val directoryConfig: DirectoryConfig, val schemaLo
       schemaUpdateStatus <- schemaLockStatus match {
         case Proceed => recreateSchema()
         case Ignore => Future.successful(Ignore)
+        case _ => Future.successful(Ignore)
```
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/schema/JndiSchemaDAO.scala:76]: this is functionally equivalent to what the code already did

**Review by dvoet — APPROVED** [review]

**Review by gpolumbo-broad — APPROVED** [review]

**Comment by MatthewBemis** [comment]
jenkins retest because elastic search couldn't come up

**Comment by MatthewBemis** [comment]
jenkins retest with leo and signinspec fixes

**Comment by MatthewBemis** [comment]
1 freetrialspec failure which is a known issue, discussed with QA and OK to merge this

**Pull request merged**